### ISSUE TITLE
(module name): (short issue description)

### ISSUE THREAD

**Issue by papiro** [aws/aws-cdk#23237]
### Describe the issue

`CodePipelineSource.gitHub` says "This is no longer the recommended method." but why not?

I tracked down the [comment in question](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/pipelines/lib/codepipeline/codepipeline-source.ts#L24) to [this PR](https://github.com/aws/aws-cdk/pull/12326) and specifically to [this commit](https://github.com/aws/aws-cdk/pull/12326/commits/6bcbe53d9eb6806079a01ffa7e6655a1944ac88c). 

The odd thing is that the comment was added with the wording as such at the time the feature was being added. But the wording seems to imply that at one point the method was the recommended method, but the method didn't exist before the PR which introduced both it and the comment. 

I am wondering why it wouldn't be a good idea to use this method, seeing as I am already using the GitHubSourceAction in one of my stacks and it is using the owner/repo/branch with the oauthToken as a SecretValue.

### Links

https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/pipelines/lib/codepipeline/codepipeline-source.ts#L24

**Comment by peterwoodworth** [comment]
This is a recommendation from CodePipeline - see [this page](https://docs.aws.amazon.com/codepipeline/latest/userguide/update-github-action-connections.html) for more info. We should expand upon this in our own docs, so thank you for bringing this up 🙂

**Comment by papiro** [comment]
oh jeez... definitely never noticed that before! Sorry about the fuss above. I was quite confused.

**Comment by kambleaa007** [comment]
is this closed or Should I Take and work on it? @papiro

**Comment by papiro** [comment]
> is this closed or Should I Take and work on it? @papiro 

I think it's just a matter of seeing if that comment is still there and figuring out why from a technical perspective it isn't the preferred way and providing the alternative way.

**Comment by gudipati** [comment]
Hi @papiro,

The documentation has been updated with the recommended alternative - https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/pipelines/lib/codepipeline/codepipeline-source.ts#L25

Does this solve your issue?

**Comment by papiro** [comment]
> Hi @papiro,
> 
> The documentation has been updated with the recommended alternative - https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/pipelines/lib/codepipeline/codepipeline-source.ts#L25
> 
> Does this solve your issue?

Yes, I believe so! I see that there is a recommendation about what to use instead, so that's great 🙂

**Comment by pahud** [comment]
> > Hi [@papiro](https://github.com/papiro),
> > The documentation has been updated with the recommended alternative - https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/pipelines/lib/codepipeline/codepipeline-source.ts#L25
> > Does this solve your issue?
> 
> Yes, I believe so! I see that there is a recommendation about what to use instead, so that's great 🙂

Awesome! Resolving now. Feel free to create a new issue if there's any concern. Cheers!
